### ISSUE TITLE
broccoli-typescript doesn't complain about errors from unchanged files during incremental compilation

### ISSUE THREAD

**Issue by IgorMinar** [angular/angular#2039]
This might be desirable when iterating on a single file, but once all the checks for that file pass, we should not go green, but instead do global check.

**Comment by caitp** [comment]
Simple approach:

1. If a previous build was a subset of the tree, and contained errors, replace `diffResult.changedPaths` with `Object.keys(this.fileRegistry);` before building

More complicated approach:
1. If a previous build was a subset of the tree and contained errors, and before that, another build was a larger subset of the tree which also contained errors, replace `diffResult.changedPaths` with `Object.keys(this.fileRegistry);` before building

The more complicated approach seems better in terms of how it works, but is probably not cheap to actually do it.

Lastly: if there are any errors, delete the output for the files containing errors. This hurts the "iterating on a single file" case, but is closer to how a typical bottom-up build system works. Or, if typescript includes something like `-Werror`, that would be good too.

**Comment by IgorMinar** [comment]
this got fixed by 2d6c44b54a3a0fd42d54f29f56f00b8db6c01b56

I went with a different approach - remember the last rebuild status, if the status was failed and the current status is pass, then do full rebuild. this is preferable to swapping paths, because doing full build via the incremental path is slow.

I thought of deleting the output files, but even if I did it it would have no effect because once the plugin throws an exception to indicate that there was a build error, the whole pipeline aborts and the output deletion doesn't make it into the dist directory.
